src/roles.js

```diff
@@ -81,9 +110,11 @@ export const NO_ROLE_CAPABILITIES = {
   },
 }
 
-/** @type {Record<RoleId, Capability>} */
-export const DEFAULT_CAPABILITIES = {
+/** @type {Record<RoleId, Role>} */
```

Comment:
```suggestion
/** @type {{ [K in RoleId]: Role<K> }} */
```

To give us a bit more type safety to avoid silly bugs in the future. Requires CREATOR_ROLE to have type `Role<typeof CREATOR_ROLE_ID>` and similarly for NO_ROLE.

Reply:
This is great; done in b8742e47508cdd69d24aeea067b38c7ee8ac3a91. Had to add 885071e93f21a8316e6d0ec17dc34d55ef3024ef as a followup to fix a type error.